### PR TITLE
Tendrl Definitions Schema 0.3

### PR DESCRIPTION
**Pull request by r0h4n**
- Bump schema version to 0.3
- Fixes Tendrl definitions validatior

### CODE REVIEW

**Comment by r0h4n** [comment]
@nthomas-redhat @TimothyAsir @shtripat @nnDarshan @GowthamShanmugam  plz review

File: tendrl/bridge_common/definitions/validator.py [hunk 1]
```diff
@@ -49,170 +24,175 @@ def response(self):
...
+    def checkAtomV3(self, atom, objects):
+        # Get the object
+        obj_name, con, atom_name = atom.split('.')[-3:]
+        obj = objects.get(obj_name, {})
         if not obj:
             return (False,
                     "object atom details not found for:%s" %
-                    (objName))
-        obj = obj.get(con, {}).get(oper, {})
+                    (obj))
```
Comment (TimothyAsir) [inline tendrl/bridge_common/definitions/validator.py]: obj -> obj_name

File: tendrl/bridge_common/definitions/validator.py [hunk 2]
```diff
@@ -49,170 +24,175 @@ def response(self):
...
+        # get the atom
+        atom = obj.get(con, {}).get(atom_name, {})
         # Check whether atoms defined
-        if not obj:
+        if not atom:
             return (False,
                     "atom:%s details not found for:%s" %
-                    (oper, objName))
-        if 'uuid' not in obj:
+                    (obj_name, atom_name))
```
Comment (TimothyAsir) [inline tendrl/bridge_common/definitions/validator.py]: (obj_name, atom_name) -> (atom_name, obj_name) ?

File: etc/samples/tendrl_definitions_generic-0.3.sample.yaml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          type: String
+        pkg_type:
+          help: "Type of package can be rpm/deb/pip/"
+        state:
+          help: "State can installed|uninstalled"
+        version:
+          help: "Version of the rpm/deb/pypi package"
+          type: String
+      enabled: true
+    Tendrl_context:
```
Comment (shtripat) [inline etc/samples/tendrl_definitions_generic-0.3.sample.yaml]: I feel for tendrl_context we should have value attribute as well

File: tendrl/bridge_common/definitions/validator.py [hunk 4]
```diff
@@ -49,170 +24,175 @@ def response(self):
...
         if 'run' not in flow:
             return (False,
                     "run not found for the flow: %s" %
-                    (flowName))
+                    (flow))
         elif flow['run'] == "":
             return (False,
                     "no value set for attribute run for the flow: %s" %
-                    (flowName))
-        if 'version' not in flow:
```
Comment (shtripat) [inline tendrl/bridge_common/definitions/validator.py]: Version field not required to be validated?

File: etc/samples/tendrl_definitions_generic-0.3.sample.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+      enabled: true
+      value: /nodes/$node.id/memory
+    Node:
+      atoms:
+        cmd:
+          enabled: true
+          inputs:
+            mandatory:
+              - Node.cmd_str
+          name: "Execute CMD on Node"
```
Comment (r0h4n) [inline etc/samples/tendrl_definitions_generic-0.3.sample.yaml]: agreed

File: etc/samples/tendrl_definitions_generic-0.3.sample.yaml (hunk 3 above)
Comment (r0h4n) [inline etc/samples/tendrl_definitions_generic-0.3.sample.yaml]: yep

File: tendrl/bridge_common/definitions/validator.py (hunk 4 above)
Comment (r0h4n) [inline tendrl/bridge_common/definitions/validator.py]: not for now, lets eval whether we really need it after T1

**Pull request merged**